syntax = "proto3";

service GrpcService {
    rpc MoveTrans(MoveTransRequest) returns (MoveTransResponse);
    rpc MoveRota(MoveRotaRequest) returns (MoveRotaResponse);
    rpc InstanceObject(InstanceObjectRequest) returns (InstanceObjectResponse);
    rpc TongsManageMove(TongsManageMoveRequest) returns (TongsManageMoveResponse);
    rpc TongsManageOpening(TongsManageOpeningRequest) returns (TongsManageOpeningResponse);
    rpc MoveCam(MoveCamRequest) returns (MoveCamResponse);
}

message MoveTransRequest {
    string dir = 1;
    int32 distance = 2;
}

message MoveTransResponse {
    string message = 1;
}

message MoveRotaRequest {
    string dir = 1;
    int32 angle = 2;
}

message MoveRotaResponse {
    string message = 1;
}

message InstanceObjectRequest {
    string object = 1;
}

message InstanceObjectResponse {
    string message = 1;
}

message TongsManageMoveRequest {
    string dir = 1;
    int32 distance = 2;
}

message TongsManageMoveResponse {
    string message = 1;
}

message TongsManageOpeningRequest {
    string move = 1;
}

message TongsManageOpeningResponse {
    string message = 1;
}

message MoveCamRequest {
    string dir = 1;
    int32 distance = 2;
}

message MoveCamResponse {
    string message = 1;
}